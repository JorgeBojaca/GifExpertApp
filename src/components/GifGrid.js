import React from 'react';
import useFetchGifs from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div>
      <h3>{category}</h3>
      {loading && 'Cargando...'}
      <div className="gif-grid-container">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
export default GifGrid;
