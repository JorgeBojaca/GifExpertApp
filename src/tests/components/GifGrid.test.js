import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas para el componente GifGrid', () => {
  test('Debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<GifGrid category="Giphy" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar correctamente los elementos', () => {
    const gifs = [
      { id: 'ABC', url: 'https://localhost/abc.jpg', title: 'LOCAL IMG' },
      { id: 'DEF', url: 'https://localhost/def.jpg', title: 'LOCAL DEF' },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category="ABC" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
