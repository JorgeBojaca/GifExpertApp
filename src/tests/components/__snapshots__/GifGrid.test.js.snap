// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente GifGrid Debe mostrar correctamente los elementos 1`] = `
<div>
  <h3>
    ABC
  </h3>
  <div
    className="gif-grid-container"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="LOCAL IMG"
      url="https://localhost/abc.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="LOCAL DEF"
      url="https://localhost/def.jpg"
    />
  </div>
</div>
`;

exports[`Pruebas para el componente GifGrid Debe mostrarse correctamente 1`] = `
<div>
  <h3>
    Giphy
  </h3>
  Cargando...
  <div
    className="gif-grid-container"
  />
</div>
`;
